### PR TITLE
HIVE-28625: Upgrade Apache Parquet version to 1.14.4

### PR DESCRIPTION
**Pull request by zhangbutao**
### Why are the changes needed?

https://github.com/apache/parquet-java/pull/3041/files fixed a bug related to data loss. We should upgrade the latest parquet version to avoid this issue.
btw, Iceberg also revert the low version 1.13.1 & upgrade to 1.14.4 to fix the issue.  https://github.com/apache/iceberg/pull/11462/files

BTW, Parquet 1.14 has added the Hadoop Vectored IO feature in https://issues.apache.org/jira/browse/PARQUET-2171, so i think we can use this good feature after the upgrading. :)

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

see the file [mvndep.log](https://github.com/user-attachments/files/17823381/mvndep.log)

### How was this patch tested?

The existing tests

### CODE REVIEW

File: iceberg/iceberg-handler/src/test/results/positive/show_create_iceberg_table.q.out [hunk 1]
```diff
@@ -345,7 +345,7 @@ TBLPROPERTIES (
   'bucketing_version'='2', 
   'current-schema'='{"type":"struct","schema-id":0,"fields":[{"id":1,"name":"i","required":false,"type":"int"},{"id":2,"name":"s","required":false,"type":"string"}]}', 
   'current-snapshot-id'='#SnapshotId#', 
-  'current-snapshot-summary'='{"added-data-files":"1","added-records":"3","added-files-size":"628","changed-partition-count":"1","total-records":"3","total-files-size":"628","total-data-files":"1","total-delete-files":"0","total-position-deletes":"0","total-equality-deletes":"0","iceberg-version":"#Masked#"}', 
+  'current-snapshot-summary'='{"added-data-files":"1","added-records":"3","added-files-size":"#Masked#","changed-partition-count":"1","total-records":"3","total-files-size":"671","total-data-files":"1","total-delete-files":"0","total-position-deletes":"0","total-equality-deletes":"0","iceberg-version":"#Masked#"}',
```
[the remark this answers is not in the document]
Comment (zhangbutao) [reply iceberg/iceberg-handler/src/test/results/positive/show_create_iceberg_table.q.out]: Good catach. Masked it. https://github.com/apache/hive/pull/5552/commits/7adde5505ff1533fe34395974358f86b91a92538

IMo, all `files size` related properties should be masked.

[the remark this answers is not in the document]
Comment (okumin) [reply iceberg/iceberg-handler/src/test/results/positive/show_create_iceberg_table.q.out]: Agree. We may finally mask all at the level of Iceberg driver or somewhere else

**Review by okumin — COMMENTED** [review]
Waiting for CI to be completed

**Review by okumin — APPROVED** [review]
LGTM!

**Pull request merged**